spec/requests/physical_switches_spec.rb

```diff
@@ -0,0 +1,100 @@
+describe "Physical Switches API" do
+  context "GET /api/physical_switch" do
+    it "returns all physical_switch" do
+      physical_switch = FactoryGirl.create(:physical_switch)
+      api_basic_authorize('physical_switch_show_list')
+
+      get(api_physical_switches_url)
+
+      expected = {
+        "name"      => "physical_switches",
+        "resources" => [{"href" => api_physical_switch_url(nil, physical_switch)}]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  context "GET /api/physical_switch/:id" do
+    it "returns one physical_switch" do
+      physical_switch = FactoryGirl.create(:physical_switch)
+      api_basic_authorize('physical_switch_show')
+
+      get(api_physical_switch_url(nil, physical_switch))
+
+      expected = {
+        "name" => physical_switch.name,
+        "href" => api_physical_switch_url(nil, physical_switch)
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  describe "Physical Switches refresh action" do
+    context "with an invalid id" do
+      it "it responds with 404 Not Found" do
+        api_basic_authorize(action_identifier(:physical_switches, :refresh, :resource_actions, :post))
+
+        post(api_physical_switch_url(nil, 999_999), :params => gen_request(:refresh))
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context "without an appropriate role" do
+      it "it responds with 403 Forbidden" do
+        physical_switch = FactoryGirl.create(:physical_switch)
+        api_basic_authorize
+
+        post(api_physical_switch_url(nil, physical_switch), :params => gen_request(:refresh))
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context "with an appropriate role" do
+      it "rejects refresh for unspecified physical switch" do
+        api_basic_authorize(action_identifier(:physical_switches, :refresh, :resource_actions, :post))
+
+        post(api_physical_switches_url, :params => gen_request(:refresh, [{"href" => "/api/physical_switches/"}, {"href" => "/api/physical_switches/"}]))
```

Comment:
maybe use api_physical_switches_url  helper method for the above 2 href.

Reply:
done